File: abc/abc169/abc169_b.fs
// https://atcoder.jp/contests/abc169/submissions/13927255

let N = stdin.ReadLine() |> int
let A = stdin.ReadLine().Split() |> Array.map int64

let rec solver prod n =
    if n = N then prod
    else
        let tmp = prod * A.[n]
        if (tmp < prod || tmp > 1000000000000000000L) then -1L
        else solver (prod * A.[n]) (n + 1)

if (Array.contains 0L A) then 0L
else solver 1L 0
|> stdout.WriteLine

File: abc/abc169/abc169_b.cs
// https://atcoder.jp/contests/abc169/submissions/13926350

using System;
using System.Linq;

class Program
{
    static void Main()
    {
        var N = int.Parse(Console.ReadLine());
        var A = Console.ReadLine().Split().Select(long.Parse).ToArray();

        if (A.Contains(0L))
        {
            Console.WriteLine(0);
            return;
        }

        var result = 1L;
        foreach (var a in A)
        {
            var temp = result * a;
            if (temp < result || temp > 1000000000000000000)
            {
                Console.WriteLine(-1);
                return;
            }
            result *= a;
        }
        Console.WriteLine(result);
    }
}
